tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -653,13 +694,13 @@ ROOT::Experimental::Detail::RPageSourceDaos::LoadClusters(std::span<RCluster::RK
    std::vector<std::unique_ptr<ROnDiskPageMapHeap>> pageMaps(clusterKeys.size());
    RDaosContainer::MultiObjectRWOperation_t readRequests;
 
-   std::size_t szPayload = 0;
+   int64_t szPayload = 0;
    unsigned nPages = 0;
 
    for (unsigned i = 0; i < clusterKeys.size(); ++i) {
       const auto &clusterKey = clusterKeys[i];
       auto clusterId = clusterKey.fClusterId;
-      std::vector<RDaosSealedPageLocator> onDiskClusterPages;
+      std::map<std::uint32_t, std::vector<RDaosSealedPageLocator>> onDiskClusterCages;
```

Comment:
The gist is that, with caging enabled, only one IOV should be assigned under each unique position being requested. Therefore, an unordered set is needed at the least, in addition to the assumption that locators are ordered as neat cage segments and checking the progression thereof in the page loop. IMO, first grouping the pages into a map of locators simplifies the code and makes it more resilient to future changes, even if it could be done without it. Lastly, this function is likely to be re-written in the upcoming backend refactor.